### DISCUSSION TITLE
How do I add a custom endpoint with DashScope?

### DISCUSSION THREAD

**Discussion by a13782425** [danny-avila/LibreChat#2529] [Q&A]
Requst
`curl --location 'https://dashscope.aliyuncs.com/api/v1/services/aigc/text-generation/generation' \
--header 'Authorization: Bearer <YOUR-DASHSCOPE-API-KEY>' \
--header 'Content-Type: application/json' \
--data '{
    "model": "llama3-8b-instruct",
    "input":{
        "messages":[
            {"content":"Where is the capital of Zhejiang?","role":"user"},
            {"content":"Thank you for asking! The capital of Zhejiang is Hangzhou.","role":"assistant"},
            {"content":"What are the interesting places there?","role":"user"}
        ]
    }
}'`
Response
`{
    "output":{
        "text":"xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx."
    },
    "usage": {
        "output_tokens": 65,
        "input_tokens": 35
    },
    "request_id":"fbd7e41a-363c-938a-81be-8ae0f9fbdb3d"
}`

How to configure？

**Comment by danny-avila** [comment]
Take a look at the other examples: https://docs.librechat.ai/install/configuration/ai_endpoints.html

**Comment by xjqianqian** [comment]
Did you succeed set the endpoint for Dashscope? Please help with my issue.

Error in docker logs:
error: [handleAbortError] AI response error; aborting request: 404 status code (no body)  

My setting in librechat.yaml:
    # Dashscope Example
    - name: 'Dashscope'
      apiKey: '${DASHSCOPE_API_KEY}'
      baseURL: 'https://dashscope.aliyuncs.com/api/v1'
      models:
        default: ['qwen-plus', 'qwen-turbo', 'qwen-max']
        fetch: false
      titleConvo: true
      titleModel: 'qwen-plus'
      modelDisplayLabel: 'Dashscope'
      iconURL: https://avatars.githubusercontent.com/u/127009744?s=200&v=4 # Add this line if you want to use a custom icon
